### PR TITLE
feat(ln-client): make LN gateway known upfront

### PR DESCRIPTION
**Pull request by elsirion**
Otherwise, fees that will be paid cannot be shown to users upfront.

### CODE REVIEW

**Review by justinmoon — APPROVED** [review]
Looks good, but seems like a lightning test is failing

Should we do the same thing to `create_bolt11_invoice`?

**Comment by elsirion** [comment]
Turns out we have tests for being able to pay internal invoices even if no LN GW is present (which is probably good). I'm trying a new, minimal approach, but longer term we need to think about how to best pull that pay fn apart so that users can be better informed.

File: gateway/ln-gateway/tests/integration_tests.rs [hunk 1]
```diff
@@ -142,12 +150,13 @@ async fn pay_valid_invoice(
...
     // User client pays test invoice
     let user_lightning_module = &user_client.get_first_module::<LightningClientModule>();
+    let gateway = user_lightning_module.select_active_gateway().await.ok();
     let OutgoingLightningPayment {
         payment_type,
         contract_id,
         fee: _,
     } = user_lightning_module
-        .pay_bolt11_invoice(invoice.clone(), ())
+        .pay_bolt11_invoice(gateway, invoice.clone(), ())
```
Comment (douglaz) [inline gateway/ln-gateway/tests/integration_tests.rs:159]: why not call `pay_invoice` here?

File: fedimint-cli/src/client.rs [hunk 2]
```diff
@@ -254,11 +254,14 @@ pub async fn handle_command(
             let lightning_module = client.get_first_module::<LightningClientModule>();
             lightning_module.select_active_gateway().await?;
 
+            let gateway = lightning_module.select_active_gateway().await.ok();
```
[the remark this answers is not in the document]
Comment (elsirion) [reply fedimint-cli/src/client.rs:257]: I was a bit unsure about this, the reason why it's an option in the first place is that we currently have support for internal payments without a gateway being present. So it wasn't entirely clear to me if thats something we'd want to warn about till it's an actual problem. But you are right that it makes debugging the root cause harder, so will add it.

File: gateway/ln-gateway/tests/integration_tests.rs (hunk 1 above)
Comment (elsirion) [reply gateway/ln-gateway/tests/integration_tests.rs]: I think I got tired of repeating the same code after a while, introduced the fn but forgot that one.

**Review by douglaz — APPROVED** [review]

**Review by okjodom — APPROVED** [review]

**Pull request merged**